stb_sprintf: error: cast from ‘char*’ to ‘unsigned int’ loses precision [-fpermissive]

stb_sprintf: error: cast from ‘char*’ to ‘unsigned int’ loses precision [-fpermissive]

reported on multiple lines of stb_sprintf.h when trying make check on tests directory,
with ppc64le machine.
```
$git clone https://github.com/nothings/stb.git stb.upstream
Cloning into 'stb.upstream'...
remote: Enumerating objects: 6545, done.
remote: Total 6545 (delta 0), reused 0 (delta 0), pack-reused 6545
Receiving objects: 100% (6545/6545), 4.41 MiB | 1.04 MiB/s, done.
Resolving deltas: 100% (4362/4362), done.
$cd stb.upstream
$gta -n2
* 052dce1 (HEAD -> master, origin/master, origin/HEAD) more fixing
* 657be5c fix README.md "noteworthy" links
$cd tests
$make
cc -I.. -Wno-pointer-to-int-cast -Wno-int-to-pointer-cast -DSTB_DIVIDE_TEST ../stb_vorbis.c test_c_compilation.c test_c_lexer.c test_dxt.c test_easyfont.c test_image.c test_image_write.c test_perlin.c test_sprintf.c test_truetype.c test_voxel.c -lm
cc -I.. -Wno-write-strings -DSTB_DIVIDE_TEST -std=c++0x test_cpp_compilation.cpp -lm -lstdc++
In file included from test_cpp_compilation.cpp:3:
../stb_sprintf.h: In function ‘int stbsp_vsprintfcb(char* (*)(char*, void*, int), void*, char*, const char*, va_list)’:
../stb_sprintf.h:323:34: error: cast from ‘const char*’ to ‘unsigned int’ loses precision [-fpermissive]
  323 |          while (((stbsp__uintptr)f) & 3) {
      |                                  ^
../stb_sprintf.h:523:35: error: cast from ‘const char*’ to ‘unsigned int’ loses precision [-fpermissive]
  523 |             if ((((stbsp__uintptr)sn) & 3) == 0)
      |                                   ^~
../stb_sprintf.h:805:38: error: cast from ‘char*’ to ‘unsigned int’ loses precision [-fpermissive]
  805 |                if ((((stbsp__uintptr)s) & 3) == 0)
      |                                      ^
../stb_sprintf.h:848:47: error: cast from ‘char*’ to ‘unsigned int’ loses precision [-fpermissive]
  848 |                         if ((((stbsp__uintptr)s) & 3) == 0)
      |                                               ^
../stb_sprintf.h:1134:44: error: cast from ‘char*’ to ‘unsigned int’ loses precision [-fpermissive]
 1134 |                      if ((((stbsp__uintptr)bf) & 3) == 0)
      |                                            ^~
../stb_sprintf.h:1172:44: error: cast from ‘char*’ to ‘unsigned int’ loses precision [-fpermissive]
 1172 |                      if ((((stbsp__uintptr)bf) & 3) == 0)
      |                                            ^~
../stb_sprintf.h:1233:38: error: cast from ‘char*’ to ‘unsigned int’ loses precision [-fpermissive]
 1233 |                if ((((stbsp__uintptr)bf) & 3) == 0)
      |                                      ^~
../stb_sprintf.h:1271:44: error: cast from ‘char*’ to ‘unsigned int’ loses precision [-fpermissive]
 1271 |                      if ((((stbsp__uintptr)bf) & 3) == 0)
      |                                            ^~
make: *** [Makefile:9: all] Error 1
```
The problem is reported while trying to build stb package for openSUSE in OBS server for ppc64/ppc64le arches:
https://build.opensuse.org/package/live_build_log/openSUSE:Factory:PowerPC/stb/standard/ppc64le

--- a/stb_sprintf.h
+++ b/stb_sprintf.h
@@ -18,6 +18,7 @@
 //    Leonard Ritter
 //    Stefano Zanotti
 //    Adam Allison
+//    Markus Kolb
 //
 // LICENSE:
 //

--- a/stb_sprintf.h
+++ b/stb_sprintf.h
@@ -200,7 +200,7 @@ STBSP__PUBLICDEF void STB_SPRINTF_DECORATE(set_separators)(char comma, char peri
 #define stbsp__uint16 unsigned short
 
 #ifndef stbsp__uintptr
-#if defined(__ppc64__) || defined(__aarch64__) || defined(_M_X64) || defined(__x86_64__) || defined(__x86_64)
+#if defined(__ppc64__) || defined(__powerpc64__) || defined(__aarch64__) || defined(_M_X64) || defined(__x86_64__) || defined(__x86_64)
 #define stbsp__uintptr stbsp__uint64
 #else
 #define stbsp__uintptr stbsp__uint32

